src/main/java/uk/nhs/tis/trainee/notifications/service/HistoryService.java

```diff
@@ -87,9 +101,56 @@ public Optional<History> updateStatus(String notificationId, NotificationStatus
       return Optional.empty();
     }
 
-    History history = optionalHistory.get();
+    return updateStatus(optionalHistory.get(), status, detail);
+  }
+
+  /**
+   * Update the status of a notification.
+   *
+   * @param traineeId      The ID of the trainee.
+   * @param notificationId The notification to update the status of.
+   * @param status         The new status.
+   * @param detail         The detail of the status.
+   * @return The updated notification history, or empty if not found.
+   */
+  public Optional<HistoryDto> updateStatus(String traineeId, String notificationId,
+      NotificationStatus status, String detail) {
```

Comment:
I like the option for detail - there's nothing that currently uses it right now, though?

Reply:
Not currently, I left it in to stay in line with the existing method I overloaded. But I'm working on some tweaks at the moment t that refactors some of this anyway.